Running jobs from bloop causes java.lang.ClassNotFoundException: com.spotify.scio.coders.WrappedBCoder

`bloop` 1.4.0

Not sure to what extent we should look into this since most of users will likely run jobs from `sbt`


```
"Error message from worker: java.lang.IllegalArgumentException: unable to deserialize com.spotify.scio.coders.WrappedBCoder
	org.apache.beam.sdk.util.SerializableUtils.deserializeFromByteArray(SerializableUtils.java:74)
	org.apache.beam.runners.dataflow.util.CloudObjectTranslators$9.fromCloudObject(CloudObjectTranslators.java:359)
	org.apache.beam.runners.dataflow.util.CloudObjectTranslators$9.fromCloudObject(CloudObjectTranslators.java:341)
	org.apache.beam.runners.dataflow.util.CloudObjects.coderFromCloudObject(CloudObjects.java:122)
	org.apache.beam.runners.dataflow.util.CloudObjectTranslators.getComponents(CloudObjectTranslators.java:72)
	org.apache.beam.runners.dataflow.util.CloudObjectTranslators.access$100(CloudObjectTranslators.java:51)
	org.apache.beam.runners.dataflow.util.CloudObjectTranslators$6.fromCloudObject(CloudObjectTranslators.java:258)
	org.apache.beam.runners.dataflow.util.CloudObjectTranslators$6.fromCloudObject(CloudObjectTranslators.java:245)
	org.apache.beam.runners.dataflow.util.CloudObjects.coderFromCloudObject(CloudObjects.java:122)
	org.apache.beam.runners.dataflow.worker.IntrinsicMapTaskExecutorFactory$2.typedApply(IntrinsicMapTaskExecutorFactory.java:345)
	org.apache.beam.runners.dataflow.worker.IntrinsicMapTaskExecutorFactory$2.typedApply(IntrinsicMapTaskExecutorFactory.java:339)
	org.apache.beam.runners.dataflow.worker.graph.Networks$TypeSafeNodeFunction.apply(Networks.java:63)
	org.apache.beam.runners.dataflow.worker.graph.Networks$TypeSafeNodeFunction.apply(Networks.java:50)
	org.apache.beam.runners.dataflow.worker.graph.Networks.replaceDirectedNetworkNodes(Networks.java:87)
	org.apache.beam.runners.dataflow.worker.IntrinsicMapTaskExecutorFactory.create(IntrinsicMapTaskExecutorFactory.java:121)
	org.apache.beam.runners.dataflow.worker.BatchDataflowWorker.doWork(BatchDataflowWorker.java:353)
	org.apache.beam.runners.dataflow.worker.BatchDataflowWorker.getAndPerformWork(BatchDataflowWorker.java:306)
	org.apache.beam.runners.dataflow.worker.DataflowBatchWorkerHarness$WorkerThread.doWork(DataflowBatchWorkerHarness.java:140)
	org.apache.beam.runners.dataflow.worker.DataflowBatchWorkerHarness$WorkerThread.call(DataflowBatchWorkerHarness.java:120)
	org.apache.beam.runners.dataflow.worker.DataflowBatchWorkerHarness$WorkerThread.call(DataflowBatchWorkerHarness.java:107)
	java.base/java.util.concurrent.FutureTask.run(FutureTask.java:264)
	java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128)
	java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628)
	java.base/java.lang.Thread.run(Thread.java:834)
Caused by: java.lang.ClassNotFoundException: com.spotify.scio.coders.WrappedBCoder
	java.base/jdk.internal.loader.BuiltinClassLoader.loadClass(BuiltinClassLoader.java:583)
	java.base/jdk.internal.loader.ClassLoaders$AppClassLoader.loadClass(ClassLoaders.java:178)
	java.base/java.lang.ClassLoader.loadClass(ClassLoader.java:521)
	java.base/java.lang.Class.forName0(Native Method)
	java.base/java.lang.Class.forName(Class.java:398)
	java.base/java.io.ObjectInputStream.resolveClass(ObjectInputStream.java:693)
	org.apache.beam.sdk.util.SerializableUtils$ContextualObjectInputStream.resolveClass(SerializableUtils.java:189)
	java.base/java.io.ObjectInputStream.readNonProxyDesc(ObjectInputStream.java:1886)
	java.base/java.io.ObjectInputStream.readClassDesc(ObjectInputStream.java:1772)
	java.base/java.io.ObjectInputStream.readOrdinaryObject(ObjectInputStream.java:2060)
	java.base/java.io.ObjectInputStream.readObject0(ObjectInputStream.java:1594)
	java.base/java.io.ObjectInputStream.readObject(ObjectInputStream.java:430)
	org.apache.beam.sdk.util.SerializableUtils.deserializeFromByteArray(SerializableUtils.java:71)
	... 23 more
```

--- a/scio-core/src/main/scala/com/spotify/scio/ScioContext.scala
+++ b/scio-core/src/main/scala/com/spotify/scio/ScioContext.scala
@@ -117,8 +117,13 @@ private object RunnerContext {
     extraLocalArtifacts: List[String]
   ): Iterable[String] = {
     val finalLocalArtifacts =
-      (detectClassPathResourcesToStage(pipelineOptions, classLoader) ++ extraLocalArtifacts)
-        .map(path => path.substring(path.lastIndexOf("/") + 1, path.length) -> path)
+      detectClassPathResourcesToStage(pipelineOptions, classLoader) ++ extraLocalArtifacts
+        .map {
+          case path if path.endsWith(".jar") =>
+            path.substring(path.lastIndexOf("/") + 1, path.length) -> path
+          case path =>
+            path -> path
+        }
         .toMap
         .values
 

